### PR TITLE
Add keyboard tests

### PR DESCRIPTION
**Pull request by MahmoudHamdy02**

### CODE REVIEW

**Review by KKoukiou — COMMENTED** [review]
I had the same issue with not being able to make the keyboard navigation to work in the test. Not sure what's the issue

File: test/check-application [hunk 1]
```diff
@@ -672,6 +672,54 @@ class TestNavigator(testlib.MachineCase):
...
+        self.restore_dir("/home/admin")
+
+        self.enter_navigator()
+
+        # FIXME: Use a loop here? (running m.execute inside for loop doesn't work)
+        m.execute("touch /home/admin/file1 /home/admin/file2 /home/admin/file3 /home/admin/file4 /home/admin/file5 /home/admin/file6 /home/admin/file7")
+        
+        # Test left/right
+        b.click("[data-item='file1']")
+        self.assertIn("pf-m-selected", b.attr("[data-item='file1']", "class"))
```
Comment (jelly) [inline test/check-application:688]: I don't think `b.attr` waits, so this will likely flake.

File: test/check-application [hunk 2]
```diff
@@ -672,6 +672,54 @@ class TestNavigator(testlib.MachineCase):
...
+        b.click("[data-item='file2']") # b.key_press(chr(39), use_ord=True)
```
Comment (jelly) [inline test/check-application:689]: The issue is here that if you add a `console.log` to `onKeyboardNav` no event shows up when you try to use `b.key_press`. So for some reason the event is not initialized or focussed.

File: test/check-application [hunk 3]
```diff
@@ -672,6 +672,54 @@ class TestNavigator(testlib.MachineCase):
...
+    def testKeyboard(self):
+        b = self.browser
+        m = self.machine
+
+        self.restore_dir("/home/admin")
+
+        self.enter_navigator()
+
+        # FIXME: Use a loop here? (running m.execute inside for loop doesn't work)
+        m.execute("touch /home/admin/file1 /home/admin/file2 /home/admin/file3 /home/admin/file4 /home/admin/file5 /home/admin/file6 /home/admin/file7")
```
Comment (jelly) [inline test/check-application:684]: you can loop in bash:

```
for i in {1..10}; do echo $i; done
```
However this is bash-ism, might not work in Ubuntu as it uses dash.

**Comment by jelly** [comment]
> I had the same issue with not being able to make the keyboard navigation to work in the test. Not sure what's the issue

Is this about ArrowUp/ArrowDown? I got delete to work with:

```python
        args = {"type": "keyDown", "code": "Delete",
                "key": "Delete", "windowsVirtualKeyCode": 46}
        self.browser.cdp.invoke("Input.dispatchKeyEvent", **args)
```

But I'm left with the issue that the browser does not seem to focus.

**Pull request closed without merging**